pyresample/geometry.py

```diff
@@ -1047,20 +1133,19 @@ def freeze(self, lonslats=None, resolution=None, shape=None, proj_info=None):
         shape = None if None in shape else shape
         area_extent = self.area_extent
         if not area_extent or not width or not height:
-            corners = self._compute_bound_centers(proj_dict, lonslats)
-            area_extent, width, height = self.compute_domain(corners, resolution, shape)
+            projection, corners = self._compute_bound_centers(proj_dict, lonslats, antimeridian_mode=antimeridian_mode)
+            area_extent, width, height = self.compute_domain(corners, resolution, shape, projection)
         return AreaDefinition(self.area_id, self.description, '',
                               projection, width, height,
                               area_extent, self.rotation)
 
-    def _compute_bound_centers(self, proj_dict, lonslats):
+    def _compute_bound_centers(self, proj_dict, lonslats, antimeridian_mode):
         lons, lats = self._extract_lons_lats(lonslats)
         if hasattr(lons, 'compute'):
-            return self._compute_bound_centers_dask(proj_dict, lons, lats)
-        return self._compute_bound_centers_numpy(proj_dict, lons, lats)
+            return self._compute_bound_centers_dask(proj_dict, lons, lats, antimeridian_mode)
+        return self._compute_bound_centers_numpy(proj_dict, lons, lats, antimeridian_mode)
 
-    def _compute_bound_centers_numpy(self, proj_dict, lons, lats):
-        # TODO: Do more dask-friendly things here
+    def _compute_bound_centers_numpy(self, proj_dict, lons, lats, antimeridian_mode):
```

Comment:
This method and `_compute_bound_centers_dask` have a lot of duplicated code. Could they be refactored to minimise the code duplication?

Reply:
Done. Let me know what you think.